src/components/Wizard/FormStep.tsx

```diff
@@ -0,0 +1,63 @@
+import { Block } from "baseui/block";
+import { Button } from "baseui/button";
+
+export function FormStep<ComponentProps = {}> ({
+  title,
+  description,
+  status = "future",
+  Component,
+}: StepProps<ComponentProps>) {
+  return null;
+};
+
+export function StepLayout<ComponentProps = {}> ({
+  previous,
+  next,
+  Component,
+  values,
+  setValues,
+  ...componentProps
+}: StepLayoutProps<ComponentProps>) {
+  const onSubmit = (arg: any) => {
+    setValues({ ...values, ...arg });
+    next();
+  };
+  return (
+    <Block
+      display="flex"
+      alignItems="flex-start"
+      justifyContent="flex-start"
+      flexDirection="column"
+    >
+      {!!previous ? (
+        <Button type="button" onClick={previous}>
+          Previous
+        </Button>
+      ) : null}
+      <Component {...componentProps as ComponentProps} handleSubmit={onSubmit} data={values} />
+    </Block>
+  );
+};
+
+export type SubStepProps = {
+  handleSubmit: (arg: any) => void;
+  data: any;
+};
+
+export type StepProps<ComponentProps = {}> = {
+  title: string;
+  hash: string;
+  description?: React.ReactNode;
+  status?: "future" | "inprogress" | "complete" | "error" | "skipped";
+  onPrevious?: (arg?: any) => void;
+  onNext?: (arg?: any) => void;
+  Component: React.FC<ComponentProps & SubStepProps>;
```

Comment:
Where do the types `React`, and `ComponentProps` get brought into scope?

Reply:
ComponentProps is passed into the step, React updated